.github/workflows/accept.yml

```diff
@@ -0,0 +1,214 @@
+      id: set-matrix
+      run: .github/scripts/matrix-from-json.sh .github/config/fallback-group.json
+
+  fallback-group-install:
+    runs-on: ubuntu-latest
+    needs: load-fallback-group
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.load-fallback-group.outputs.matrix)}}
+    container:
+        image: ${{ matrix.image }}
+
+    env:
+      # TODO: move to staging when AO-20147
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }}
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Show Environment Info
+        run: |
+          printenv
+          node --version
+          npm --version 
+          cat /etc/os-release
+
+      # the staging bucket has been cleared
+      # install of package will fallback to build from source
+      - name: NPM Install (to fallback)
+        run: npm install --unsafe-perm --s3_host=staging
+
+      - name: Check Artifacts
+        run: ls ./dist/napi-v*/apm_bindings.node && ls ./dist/napi-v*/ao_metrics.node
+
+      - name: Run tests
+        run: npm test
+        if: ${{ github.event.inputs.target-test }}
+
+  build-group-publish:
+    runs-on: ubuntu-latest 
+    needs: [load-build-group, fallback-group-install]
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.load-build-group.outputs.matrix)}}
+    container:
+        image:  ${{ matrix.image }}
+
+    env: 
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Show Environment Info
+        run: |
+          printenv
+          node --version
+          npm --version 
+          cat /etc/os-release
+
+      # must install specific dependencies before a build
+      # can't call npm install. doing so may fallback-to-build if package has yet to be published (double build)
+      # use npm workaround specifying a package name to bypass install script in package.json
+      - name: NPM Install dependencies
+        run: npm install linux-os-info --unsafe-perm
+
+      # must setup libobo before build
+      # node-pre-gyp rebuild runs "clean" and "build" at once.
+      # it is mapped to `node-gyp rebuild` which internally means "clean + configure + build" and triggers a full recompile
+      - name: Build
+        run: |
+          node setup-liboboe.js
+          npx node-pre-gyp rebuild
+
+      # artifacts are at:build/stage/nodejs/bindings/
+      - name: Package
+        run: npx node-pre-gyp package # requires clean rebuild
+
+      # *** IMPORTANT: 
+      # always include --s3_host flag regardless of node-pre-gyp defaults.
+      # node-pre-gyp can't publish over existing version, thus always attempt unpublished of current in staging bucket
+      - name: Clear Staging for version
+        run: npx node-pre-gyp unpublish --s3_host=staging
```

Comment:
Oh I take my above comment about fail fast back -- staging is specifically cleared again here.  I guess that is the safest approach and doesn't require knowing that the `build-group-unpublish` job is indirectly chained as a dependency.  Actually just to confirm, the "needs:" chaining ensures that `build-group-unpublish` must always run first before this `build-group-publish` can run, right?

Reply:
Comment above was on spot. Not taking it back ;) We should not get here if *any* in `build-group-unpublish`. This step should have been removed when job chaining was modified.

Reply:
👍

Reply:
Done. Workflow ran in dev as expected.